var/spack/repos/builtin/packages/cray-libsci/package.py

```diff
@@ -55,16 +56,19 @@ def blas_libs(self):
         shared = True if "+shared" in self.spec else False
         compiler = self.spec.compiler.name
 
+        lib = []
         if "+openmp" in self.spec and "+mpi" in self.spec:
-            lib = "libsci_{0}_mpi_mp"
+            lib = ["libsci_{0}_mpi_mp", "libsci_{0}_mp"]
         elif "+openmp" in self.spec:
-            lib = "libsci_{0}_mp"
+            lib = ["libsci_{0}_mp"]
         elif "+mpi" in self.spec:
-            lib = "libsci_{0}_mpi"
+            lib = ["libsci_{0}_mpi", "libsci_{0}"]
         else:
-            lib = "libsci_{0}"
+            lib = ["libsci_{0}"]
 
-        libname = lib.format(self.canonical_names[compiler].lower())
+        libname = []
+        for i in range(len(lib)):
+            libname.append(lib[i].format(self.canonical_names[compiler].lower()))
```

Comment:
What about avoiding use of an index with something like:

```suggestion
        for lib_fmt in lib:
            libname.append(lib_fmt.format(self.canonical_names[compiler].lower()))
```